internal/otg_helpers/otg_config_helpers/otgflowhelpers.go

```diff
@@ -260,11 +262,17 @@ func (f *Flow) AddIPv6Header() {
 	ipv6Hdr := f.flow.Packet().Add().Ipv6()
 	if f.IPv6Flow.IPv6SrcCount != 0 {
 		ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount)
+		if f.IPv6Flow.IPv6SrcStep != "" {
+			ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount).SetStep(f.IPv6Flow.IPv6SrcStep)
+		}
 	} else {
 		ipv6Hdr.Src().SetValue(f.IPv6Flow.IPv6Src)
 	}
 	if f.IPv6Flow.IPv6DstCount != 0 {
 		ipv6Hdr.Dst().Increment().SetStart(f.IPv6Flow.IPv6Dst).SetCount(f.IPv6Flow.IPv6DstCount)
+		if f.IPv6Flow.IPv6DstStep != "" {
+			ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount).SetStep(f.IPv6Flow.IPv6DstStep)
+		}
 	} else {
 		ipv6Hdr.Dst().SetValue(f.IPv6Flow.IPv6Dst)
 	}
```

Comment:
https://go.dev/doc/effective_go#switch

Reading nested `if/else` blocks is hard to read.

Converting the outer `if/else` to a `switch` improves the readability.


```suggestion
switch {
case f.IPv6Flow.IPv6SrcCount != 0:
		ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount)
		if f.IPv6Flow.IPv6SrcStep != "" {
			ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount).SetStep(f.IPv6Flow.IPv6SrcStep)
		}
default:
		ipv6Hdr.Src().SetValue(f.IPv6Flow.IPv6Src)
}

switch }
case f.IPv6Flow.IPv6DstCount != 0:
		ipv6Hdr.Dst().Increment().SetStart(f.IPv6Flow.IPv6Dst).SetCount(f.IPv6Flow.IPv6DstCount)
		if f.IPv6Flow.IPv6DstStep != "" {
		 	ipv6Hdr.Src().Increment().SetStart(f.IPv6Flow.IPv6Src).SetCount(f.IPv6Flow.IPv6SrcCount).SetStep(f.IPv6Flow.IPv6DstStep)
		}
default:
		ipv6Hdr.Dst().SetValue(f.IPv6Flow.IPv6Dst)
}
```